packages/core/database/src/metadata/relations.ts

```diff
@@ -1,26 +1,69 @@
-/**
- * @module relations
- */
+import _ from 'lodash/fp';
+
+import type { Meta, Metadata } from './metadata';
+import type { RelationalAttribute, Relation, MorphJoinTable } from '../types';
+
+interface JoinColumnOptions {
+  attribute: (Relation.OneToOne | Relation.ManyToOne) & Relation.Owner;
+  attributeName: string;
+  meta: Meta;
+}
+
+interface JoinTableOptions {
+  attribute: RelationalAttribute & Relation.WithTarget & Relation.Bidirectional;
+  attributeName: string;
+  meta: Meta;
+}
+
+const hasInversedBy = (
```

Comment:
Just a note, I've realized a lot of this is duplicated by the Attribute relations types, but we can reconcile that later to have a single source of truth for these properties.